project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module

```diff
@@ -2181,44 +2205,179 @@ function c4m_og_get_group_status($group) {
+  // if 'editable by members' option is enabled for this page.
+  if ($op == 'update' && ($group = c4m_og_get_group_of_content($node)) && $node->type == 'wiki_page' && $node->status == NODE_PUBLISHED) {
+    if (_c4m_features_og_members_get_user_type($group) == 'member' && c4m_content_wiki_page_is_editable_by_members($node)) {
+      return NODE_ACCESS_ALLOW;
     }
   }
 
-  // Grant access to group content by the same access to the group.
-  if (is_object($node) && og_is_group_content_type('node', $node->type)) {
+  // When $node is string (happens when creating content), it contains the
+  // type of created content.
+  $node_type = is_string($node) ? $node : $node->type;
+
+  // Handling group content bundles.
+  if (og_is_group_content_type('node', $node_type)) {
+
+    if ($op == 'view') {
+      // Couldn't resolve the group of the content.
+      if (!$group = c4m_og_get_group_of_content($node)) {
+        return NODE_ACCESS_IGNORE;
+      }
+
+      // Group content is saved as draft.
+      if ($node->status == NODE_NOT_PUBLISHED) {
+
+        // Only power user or content author may view the node.
+        if (_c4m_features_og_members_is_power_user($group) || $node->uid == $user->uid) {
+          return NODE_ACCESS_ALLOW;
+        }
+        else {
+          return NODE_ACCESS_DENY;
+        }
+      }
 
-    // Couldn't resolve the group of the content.
-    if (!$group = c4m_og_get_group_of_content($node)) {
-      return NODE_ACCESS_IGNORE;
+      $group_access = node_access($op, $group, $account);
+
+      return $group_access ? NODE_ACCESS_ALLOW : NODE_ACCESS_DENY;
     }
+    // Handling Create / Update / Delete operations.
+    else {
+
+      $user_type = _c4m_features_og_members_get_user_type();
+      // Can't resolve user type - couldn't get OG context to resolve group.
+      if ($user_type == 'unknown') {
+        return NODE_ACCESS_IGNORE;
+      }
+
+      // Visitors are not allowed to Create / Update / Delete content.
+      if ($user_type == 'visitor') {
+        return NODE_ACCESS_DENY;
+      }
 
-    if (c4m_og_get_group_status($group) != 'pending') {
-      return NODE_ACCESS_IGNORE;
+      // We allow access only for draft / published groups.
+      $group = c4m_og_current_group();
+      $group_state = c4m_og_get_group_status($group);
+      $allowed_states = array('draft', 'published');
+
+      // Can't resolve group state, or group state is not allowed.
+      if (!$group_state || !in_array($group_state, $allowed_states)) {
+        return NODE_ACCESS_DENY;
+      }
+
+      // We allow access to power users, any member, when creating, and member
+      // that is also content author, when editing or deleting.
+      $allowed_user_types = array('admin', 'owner', 'site-admin');
+      if (in_array($user_type, $allowed_user_types)) {
+        return NODE_ACCESS_ALLOW;
+      }
+      elseif ($user_type == 'member') {
+
+        if ($op == 'create') {
```

Comment:
we're probably missing the create wiki_page by member (which should be allowed to power users only)